branding_company/model/stock_picking.py

```diff
@@ -26,19 +25,31 @@ class StockPicking(models.Model):
     """Extend stock.picking with branding."""
     _inherit = 'stock.picking'
 
-    def _prepare_invoice(
-            self, cr, uid, picking, partner, inv_type, journal_id,
-            context=None):
+    def _compute_branding(self):
+        """Branding for a delivery order is taken from sales order.
+
+        We use the procurement group to find all relates sale orders.
+        The first sale order found will determine the branding.
+        """
+        sale_model = self.env['sale.order']
+        sale_orders = sale_model.search([
+            ('procurement_group_id', '=', self.group_id)])
+        return (
+            sale_orders and
+            sale_orders[0].branding_company and
+            sale_orders[0].branding_company.id or False
```

Comment:
I heard a talk where Anthony was proud how concise code you can write in the new api, but it seems silly to look that up now. Guarantees in the code:
https://github.com/OCA/OCB/blob/8.0/openerp/models.py#L5574
https://github.com/OCA/OCB/blob/8.0/openerp/fields.py#L802
https://github.com/OCA/OCB/blob/8.0/openerp/fields.py#L1482
try it and be amazed